File: Decorator/C#/CoffeeShopDecorator/MilkCondiment.cs
namespace CoffeeShopDecorator
{
    //ConcreteDecorator
    public class MilkCondiment : CondimentDecorator
    {
        public MilkCondiment(Beverage beverage) : base(beverage)
        {
            Description = "milk";
        }
        public override string GetDescription()
        {
            return $"{base.GetDescription()}, {Description}";
        }

        public override decimal GetPrice()
        {
            return base.GetPrice() + 0.2m;
        }
    }
}

File: Decorator/Java/CoffeeShowDecorator/src/CoffeeShopDecorator/MilkCondiment.java
package CoffeeShopDecorator;

//ConcreteDecorator
public class MilkCondiment extends CondimentDecorator
{
    public MilkCondiment(Beverage beverage) 
    {
    	super(beverage);
        description = "milk";
    }

    @Override
    public double getPrice()
    {
        return wrappedBeverage.getPrice() + 0.2d;
    }

    @Override
    public String getDescription()
    {
        return String.format("%s, %s", wrappedBeverage.getDescription(), description);
    }
}